Notebooks/Exploration.ipynb

```diff
@@ -573,6 +576,113 @@
     "SK_model.plot(SK_forecast, xlabel = 'Date', ylabel = 'Scaled Infections')\n",
     "plt.title('Infections for South Korea');"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Weather Data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "frequency = 24\n",
+    "start_date = '01-JAN-2020'\n",
+    "end_date = '28-MAR-2020'\n",
+    "api_key = 'f2c76ab75910443592c104308202903'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "\n",
+      "Retrieving weather data for Germany\n",
+      "\n",
+      "\n",
+      "Currently retrieving data for Germany: from 2020-01-01 to 2020-01-31\n",
+      "Time elapsed (hh:mm:ss.ms) 0:00:01.120985\n",
+      "Currently retrieving data for Germany: from 2020-02-01 to 2020-02-29\n",
+      "Time elapsed (hh:mm:ss.ms) 0:00:02.419707\n",
+      "Currently retrieving data for Germany: from 2020-03-01 to 2020-03-28\n",
+      "Time elapsed (hh:mm:ss.ms) 0:00:12.524671\n",
+      "\n",
+      "\n",
+      "export Germany completed!\n",
+      "\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Germany\n",
+    "location_list = ['Germany']\n",
+    "os.chdir(\"../Data/weather/germany\")\n",
+    "\n",
+    "hist_weather_data = retrieve_hist_data(api_key,\n",
+    "                                location_list,\n",
+    "                                start_date,\n",
+    "                                end_date,\n",
+    "                                frequency,\n",
+    "                                location_label = False,\n",
+    "                                export_csv = True,\n",
+    "                                store_df = True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "\n",
+      "Retrieving weather data for mongolia\n",
+      "\n",
+      "\n",
+      "Currently retrieving data for mongolia: from 2020-01-01 to 2020-01-31\n",
+      "Time elapsed (hh:mm:ss.ms) 0:00:01.252103\n",
+      "Currently retrieving data for mongolia: from 2020-02-01 to 2020-02-29\n",
+      "Time elapsed (hh:mm:ss.ms) 0:00:02.467010\n",
+      "Currently retrieving data for mongolia: from 2020-03-01 to 2020-03-28\n",
+      "Time elapsed (hh:mm:ss.ms) 0:00:03.557718\n",
+      "\n",
+      "\n",
+      "export mongolia completed!\n",
+      "\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "# China\n",
+    "location_list = STATES_CH\n",
```

Comment:
Not all states can be retrieved. Hong Kong and inner Mongolia need to be retrieved manually.